app/backend/src/couchers/servicers/auth.py

```diff
@@ -48,42 +49,65 @@ def get_session_for_token(self, token):
         """
         Returns None if the session token is not valid, and (user_id, jailed) corresponding to the session token otherwise.
 
-        TODO(aapeli): session expiry
+        Also updates the token.
         """
         with session_scope() as session:
             result = (
                 session.query(User, UserSession)
                 .join(User, User.id == UserSession.user_id)
                 .filter(UserSession.token == token)
+                .filter(UserSession.is_valid)
                 .one_or_none()
             )
 
             if not result:
                 return None
             else:
-                return result.User.id, result.User.is_jailed
+                user, user_session = result
 
-    def _create_session(self, context, session, user):
+                # let's update the token
+                user_session.last_seen = func.now()
```

Comment:
Update `user.last_active' here as well, so we can get rid of `UpdateLastActiveTimeInterceptor`.

Reply:
I agree. Can you think of a better name than `get_session_for_token`? It's not particularly descriptive

Reply:
Or maybe we don't want to update `user.last_active` for automatic background pings that just keep the session alive?

Reply:
`bump_session_and_get_user` maybe. I'm not good at naming either, but the docstring is good so naming is not super important.

Reply:
> Or maybe we don't want to update `user.last_active` for automatic background pings that just keep the session alive?

At the moment both interceptors run for the same queries, so it's all the same. This may need some more granularity once we start having mobile apps where calling the API does not imply being active (e.g. fetching notification info does not make one active). For now though, I do think these should be combined.